docs/source/concepts/interoperability.rst

```diff
@@ -0,0 +1,474 @@
+  +---------------+-----------------+-----------------+
+  | **Transfer**  | maybe           | maybe           |
+  +---------------+-----------------+-----------------+
+
+  For non-consuming **Exercise** and **Fetch** actions, the incoming ledger must be the same as the outgoing ledger.
+  **Transfer** actions must have at least one of them.
+  A **transfer** action with both set represents a complete transfer. If only the incoming ledger is set, that represents the partial information of an **Enter** event, if only outgoing is set, that's the partial information of a **Leave** event.
+
+The :ref:`action order <def-action-order>` generalizes to interoperable causality graphs accordingly.
+
+In the :ref:`example for Enter and Leave events <enter-leave-event>` where the painter exercises three choices on contract `c` with signatories Alice and the painter, the four transactions yield the following interoperable causality graph.
+Incoming and outgoing ledgers are encoded as colors (green for Ledger 1 and yellow for Ledger 2).
+**Transfer** vertices are shown as circles, where the left half is colored with the incoming ledger and the right half with the outgoing ledger.
+
+.. https://app.lucidchart.com/documents/edit/ef1e60ac-fa1e-40be-b1e6-7b3197d4543b
+
+.. _interoperable-causality-graph-linear:
+   
+.. figure:: ./images/interoperable-causality-graph-linear.svg
+   :align: center
+   :width: 100%
+
+   Interoperable causality graph with transfer actions
+
+.. note::
+   As for ordinary causality graphs, the diagrams for interoperable causality graphs omit transitive edges for readability.
+
+As an example for a cross-domain transaction, consider the :ref:`paint counteroffer workflow with the cross-domain transaction <cross-ledger-transaction>`.
+The corresponding interoperable causality graph is shown below.
+The last transaction `tx4` is a cross-ledger transaction because its actions have more than one color.
+
+.. https://app.lucidchart.com/documents/edit/c3b120cf-1974-4ae8-8334-435642f94eed/
+
+.. _counteroffer-interoperable-causality-graph:
+   
+.. figure:: ./images/counteroffer-interoperable-causality-graph.svg
+   :align: center
+   :width: 100%
+
+   Interoperable causality graph for the paint counteroffer workflow on two DAML ledgers
+
+
+Consistency
+===========
+
+Definition »Ledger trace«
+  A **ledger trace** is a finite list of pairs `(a`:sub:`i`\ `, b`:sub:`i`\ `)`
+  such that `b`:sub:`i - 1` = `a`:sub:`i` for all `i` > 0.
+  Here `a`:sub:`i` and `b`:sub:`i` identify DAML ledgers or are the special value `NONE`,
+  which is different from all DAML ledger identifiers.
+
+ 
+Definition »Interoperable causal consistency for a contract«
+  Let `G` be an interoperable causality graph and `X` be a set of actions from `G` on a contract in `c`.
+  The graph `G` is **interoperable consistent for the contract** `c` on `X` if all of the following hold:
+
+  #. If `X` is not empty, then `X` contains a **Create** or **Enter** action.
```

Comment:
`Enter` hasn't really been defined as an action up until now (they're mentioned as events before).

Reply:
True. I used to define them, but @bame-da's rephrasing removed them as actions. Added back in.